### PR TITLE
Add arm64-macos editor build

### PR DESCRIPTION
**Pull request by vlaaad**
User-facing changes: 
This changeset adds support for the arm64 macOS version of the editor.

Technical notes:
I changed the editor `build` and `bundle` steps. 
Before: `build` prepares and compiles code, then creates an uberjar with deps for all platforms, `bundle` uses jar and platform JDK to create platform-dependent executable.
After: `build` prepares and compiles code, `bundle` creates uberjar with deps for the target platform, then uses jar and platform JDK to create platform-dependent executable.

Related to #7781

### CODE REVIEW

File: editor/src/java/com/defold/editor/providers/JsonProviders.java [hunk 1]
```diff
@@ -127,15 +127,15 @@ private static Object JsonToObject(JsonNode node, Message.Builder builder, Field
...
-                    return node.getValueAsDouble();
+                    return node.asDouble();
 
             case VALUE_NUMBER_INT:
                 if (type == FieldDescriptor.JavaType.INT)
-                    return node.getValueAsInt();
+                    return node.asInt();
                 else if (type == FieldDescriptor.JavaType.LONG)
-                    return node.getValueAsLong();
+                    return node.asLong();
```
Comment (vlaaad) [inline editor/src/java/com/defold/editor/providers/JsonProviders.java:138]: Fixed deprecation warnings after adding `:javac-options ["-Xlint:unchecked" "-Xlint:deprecation"]` to `project.clj`

File: editor/src/java/com/dynamo/upnp/SSDP.java [hunk 2]
```diff
@@ -46,7 +38,7 @@ public boolean connect(NetworkInterface networkInterface) throws IOException {
             if (!addresses.isEmpty()) {
                 mcastSocket = new MulticastSocket(SSDP_MCAST_PORT);
                 mcastSocket.setNetworkInterface(networkInterface);
-                mcastSocket.joinGroup(SSDP_MCAST_ADDR);
+                mcastSocket.joinGroup(new InetSocketAddress(SSDP_MCAST_ADDR, 0), null);
```
Comment (vlaaad) [inline editor/src/java/com/dynamo/upnp/SSDP.java:41]: Fixed another deprecation warning. JDK's javadocs contained this snippet with the equivalent undeprecated code.

**Review by matgis — APPROVED** [review]
LGTM!

File: editor/tasks/leiningen/pack.clj [hunk 3]
```diff
@@ -57,11 +59,16 @@
    "${DYNAMO-HOME}/ext/bin/x86_64-macos/luajit-32"           "x86_64-macos/bin/luajit-32"
    "${DYNAMO-HOME}/ext/bin/x86_64-macos/luajit-64"           "x86_64-macos/bin/luajit-64"
 
+   "${DYNAMO-HOME}/ext/bin/arm64-macos/luajit-32"            "arm64-macos/bin/luajit-32"
+   "${DYNAMO-HOME}/ext/bin/arm64-macos/luajit-64"            "arm64-macos/bin/luajit-64"
```
Comment (JCash) [inline editor/tasks/leiningen/pack.clj:63]: At some point, we should use the unpack folder from bob.jar

**Review by JCash — APPROVED** [review]
LGTM!

**Pull request merged**